### PR TITLE
fix(java): check if a version exists when determining GAV by file name for `jar` files

### LINKED ISSUE #5627: identify jar files by file names only for GAV from required version
## Description
When we determinate jar package by file name (https://github.com/aquasecurity/trivy/blob/ad977a4256920d6269e84c82e4f42bbe244a2c5a/pkg/javadb/client.go#L145-L176) - we check only number of indexes.
But we also should check that required version (version from file name) exists.

Discussion  about this case - #5577

Example:
there is `elasticsearch-7.10.2.jar` file  built from source (sha1 for file != sha1 from trivy-java-db).
So we try to find GroupID by ArtifactID.
Trivy currently takes GroupID with maximum number of versions:
```bash
POM was determined in a heuristic way	{"file": "elasticsearch-7.10.2.jar", "artifact": "software.amazon.awssdk:elasticsearch:7.10.2"}
```

But `software.amazon.awssdk:elasticsearch` doesn't have 7.10.2 version - https://mvnrepository.com/artifact/software.amazon.awssdk/elasticsearch

### PR DESCRIPTION
**Pull request by DmitriyLewen**
## Description
See #5627

## Related issues
- Close #5627

## Related PRs
- [x] aquasecurity/go-dep-parser/pull/276
- [x] aquasecurity/trivy-java-db/pull/23

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

File: pkg/javadb/client.go [hunk 1]
```diff
@@ -142,37 +142,49 @@ func (d *DB) SearchBySHA1(sha1 string) (jar.Properties, error) {
 	}, nil
 }
 
-func (d *DB) SearchByArtifactID(artifactID string) (string, error) {
+func (d *DB) SearchByArtifactID(artifactID, version string) (string, error) {
 	indexes, err := d.driver.SelectIndexesByArtifactIDAndFileType(artifactID, types.JarType)
```
Comment (knqyf263) [inline pkg/javadb/client.go:146]: What if changing this method to match the version as well as the artifact ID?

Comment (DmitriyLewen) [reply pkg/javadb/client.go:146]: Do you mean to move the following logic to "trivy-java-db"?
https://github.com/aquasecurity/trivy/blob/2f10ed1c360dd03061481eee97be8c66bd9d5ccc/pkg/javadb/client.go#L157-L188

Or update our sql query in "trivy-java-db"?

Comment (knqyf263) [reply pkg/javadb/client.go:146]: I meant SQL query. Is it possible?

Comment (DmitriyLewen) [reply pkg/javadb/client.go]: I created https://github.com/aquasecurity/trivy-java-db/pull/23 and updated this PR.
Take a look, when you have time, please.

**Comment by DmitriyLewen** [comment]
@knqyf263 i updated `trivy-java-db` version.
Take a look this PR please.

**Review by knqyf263 — APPROVED** [review]

**Pull request merged**